src/main/scala/org/broadinstitute/dsde/firecloud/service/ExportEntitiesByTypeActor.scala

```diff
@@ -1,79 +1,260 @@
-
-import akka.actor.{Actor, Props}
+import akka.Done
+import akka.actor.{Actor, ActorContext, ActorRef, Props}
 import akka.pattern.pipe
-import org.broadinstitute.dsde.firecloud.Application
+import akka.stream._
+import akka.stream.scaladsl._
+import akka.util.{ByteString, Timeout}
+import better.files.File
+import com.typesafe.scalalogging.slf4j.LazyLogging
 import org.broadinstitute.dsde.firecloud.dataaccess.RawlsDAO
-import org.broadinstitute.dsde.firecloud.model.{ModelSchema, UserInfo}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.broadinstitute.dsde.firecloud.model.{UserInfo, _}
 import org.broadinstitute.dsde.firecloud.service.ExportEntitiesByTypeActor.ExportEntities
-import org.broadinstitute.dsde.firecloud.service.PerRequest.{PerRequestMessage, RequestCompleteWithHeaders}
-import org.broadinstitute.dsde.firecloud.utils.TSVFormatter
-import spray.http.MediaTypes._
-import spray.http.StatusCodes._
-import spray.http._
+import org.broadinstitute.dsde.firecloud.utils.StreamingActor.{FirstChunk, NextChunk}
+import org.broadinstitute.dsde.firecloud.utils.{StreamingActor, TSVFormatter}
+import org.broadinstitute.dsde.firecloud.{Application, FireCloudConfig, FireCloudExceptionWithErrorReport}
+import org.broadinstitute.dsde.rawls.model._
+import spray.http.{ContentTypes, _}
+import spray.json._
+import spray.routing.RequestContext
 
+import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
-import scala.util.Success
+import scala.language.postfixOps
+
 
 object ExportEntitiesByTypeActor {
+
   sealed trait ExportEntitiesByTypeMessage
-  case class ExportEntities(workspaceNamespace: String, workspaceName: String, filename: String, entityType: String, attributeNames: Option[IndexedSeq[String]]) extends ExportEntitiesByTypeMessage
+
+  case class ExportEntities(ctx: RequestContext, workspaceNamespace: String, workspaceName: String, entityType: String, attributeNames: Option[IndexedSeq[String]]) extends ExportEntitiesByTypeMessage
 
   def props(exportEntitiesByTypeConstructor: UserInfo => ExportEntitiesByTypeActor, userInfo: UserInfo): Props = {
     Props(exportEntitiesByTypeConstructor(userInfo))
   }
 
-  def constructor(app: Application)(userInfo: UserInfo)(implicit executionContext: ExecutionContext) =
-    new ExportEntitiesByTypeActor(app.rawlsDAO, userInfo)
+  def constructor(app: Application, materializer: ActorMaterializer)(userInfo: UserInfo)(implicit executionContext: ExecutionContext) =
+    new ExportEntitiesByTypeActor(app.rawlsDAO, userInfo, materializer)
 }
 
-class ExportEntitiesByTypeActor(val rawlsDAO: RawlsDAO, val userInfo: UserInfo)(implicit protected val executionContext: ExecutionContext) extends Actor with ExportEntitiesByType {
+class ExportEntitiesByTypeActor(val rawlsDAO: RawlsDAO, val argUserInfo: UserInfo, argMaterializer: ActorMaterializer)(implicit protected val executionContext: ExecutionContext) extends Actor with LazyLogging {
```

Comment:
I'd love to simplify this, but I can't get the syntax/compilation to work right. Making those two vals implicit breaks the constructor in the companion object. Will look for a way to make this work.